### DISCUSSION TITLE
Windows 7 SP 1 - Make a GET via https

### DISCUSSION THREAD

**Discussion by inverplax** [wqweto/VbAsyncSocket#45] [General]
Hi,
please, is it possible with your project to make a GET via https and get the response?
This should be working also on Windows 7 SP1 32bit. 

Thank you in advance,
Inverplax

**Comment by inverplax** [comment]
Hi,
I tried this code, but I get an error.

Dim HTTPRequest As cHttpRequest


    Set HTTPRequest = New cHttpRequest
    With HTTPRequest
        
        .Option_(WinHttpRequestOption_SecureProtocols) = SecureProtocol_ALL Or SecureProtocol_TLS1_1 Or SecureProtocol_TLS1_2
        .Option_(WinHttpRequestOption_EnableHttpsToHttpRedirects) = True
        .Option_(WinHttpRequestOption_SslErrorIgnoreFlags) = SslErrorFlag_Ignore_All
        .Open_ "GET", HTTPS, False        
        .Send        
       
        Debug.Print .ResponseText
               
    End With


Please, note that instead of HTTPS I have a valid website who will answer with a sequence of characters and numbers.

The error I get is 10035,  "it's impossible to complete the operation immediately on the socket not in chunks.." 
Please, How can I do to get the answer of my website?

Regards,

**Comment by wqweto** [comment]
> How can I do to get the answer of my website?

Post your actual URL if you need me to take a look what's happening with your server exactly.

**Reply by inverplax** [reply]
Hi Vladimir,

thank you! Try this https://www.prolocobaldissero.it/tuomenu/order/9L1TB or https://www.prolocobaldissero.it/tuomenu/order/NS938

**Reply by wqweto** [reply]
Yes, this seems to be a problem with cHttpRequest which I just fixed in commit 3935cfcb4823b30a2f1f0571facc683f8581afa3

Try to download fixed `HttpRequest.zip` version 1.0.6 once again and check if it's working under your Win7

**Comment by inverplax** [comment]
Hi Vladimir,
yes, it's working now. Thank you very much!
Do you think that this solution can be working also on xp?

Regards,

**Reply by wqweto** [reply]
Definitely. Your code should work ok under NT 4.0 onwards (incl. Win2000 and WinXP) if my tests here are worth anything.
